lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -202,9 +214,13 @@ def _loop_process_output(self, output):
         if not self.seen_output:
             msg = check_output(time_passed, self.output_timeout)
             if msg:
-                self.log.warn(msg)
                 # avoid getting warning multiple times by setting seen_output to True if a warning was produced
                 self.seen_output = True
+                if self.fatal_option:
+                    self.stop_tasks()
+                    raise RunMPIException(TIMEOUT_EXIT_CODE, fatal_msg(msg))
```

Comment:
I would just use `self.log.error` and exit (raising an error means trowing a traceback showing the guts of the code, which is not useful to users).
Also, always do the `self.log.warn` with the standard message, and just emit an additional error message in case fatal is enabled:

```python
self.log.warn(msg)
if self.fatal_option:
    self.stop_tasks()
    error_msg = "No output detected after %d seconds" % ...
    error_msg += "; this is considered fatal (unless --disable-output-check-fatal is used)"
    self.log.error(error_msg)
    sys.exit(TIMEOUT_EXIT_CODE)
```